### ISSUE TITLE
datetime64/timedelta64 do not accept numpy integer types

### ISSUE THREAD

**Issue by damiendr** [numpy/numpy#10004]
Example:
```python
>>> np.datetime64(1478476800, "s")
numpy.datetime64('2016-11-07T00:00:00')

>>> np.datetime64(np.int64(1478476800), "s")
ValueError: Could not convert object to NumPy datetime

>>> np.timedelta64(1, "s")
numpy.timedelta64(1,'s')

>>> np.timedelta64(np.int64(1), "s")
ValueError: Could not convert object to NumPy timedelta
```

**Comment by shoyer** [comment]
Agreed, it would be nice to support this!

**Comment by eric-wieser** [comment]
Should be a straightforward change in `convert_pyobject_to_datetime` to replace `PyInt_Check(obj) || PyLong_Check(obj)` with a `PyArray_PyIntAsIntp(x)`

**Comment by matveich** [comment]
@eric-wieser I've changed the code as you suggested and it works. However, I have some troubles with tests: when assert_equal compares 2 datetime64 objects, it returns OK even if these objects aren't equal. Moreover, when I change old tests to wrong, they still pass. So, in my tests I compare only strings

**Comment by matveich** [comment]
Of course, it can be something wrong with my python environment, but I don't know what

**Comment by shoyer** [comment]
@matveich that looks like another bug, so I opened #10081.

In the meantime, it would be fine to use `self.assertEqual` or `np.testing.assert_` (with a comment noting GitHub issue 10081).

**Comment by mjpietrzak** [comment]
More than 3 years have passed, yet NumPy still does not accept its own integer type...

**Comment by Ijtihed** [comment]
the issue still there seems to be the `datetime64` constructor path: `timedelta64(np.int64(...), unit)` already works but `datetime64(np.int64(...), unit)` still falls through

I believe this fixes it well
